src/Resources/Proxy/ShipmentProxy.php

```diff
@@ -18,774 +18,506 @@
     }
 
     /**
-     * @param string $recipientTaxNumber
-     * @return $this
-     * @deprecated
+     * @deprecated Use setRecipientTaxIdentificationNumbers() or addRecipientTaxIdentificationNumber() instead.
      */
-    public function setRecipientTaxNumber($recipientTaxNumber)
+    public function setRecipientTaxNumber(?string $recipientTaxNumber): self
     {
         $this->getResource()->setRecipientTaxNumber($recipientTaxNumber);
 
         return $this;
     }
 
     /**
-     * @return string|null
-     * @deprecated
+     * @deprecated Use getRecipientTaxIdentificationNumbers() instead.
      */
-    public function getRecipientTaxNumber()
+    public function getRecipientTaxNumber(): ?string
     {
         return $this->getResource()->getRecipientTaxNumber();
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function setRecipientTaxIdentificationNumbers(array $taxIdentificationNumbers)
+    public function setRecipientTaxIdentificationNumbers(array $taxIdentificationNumbers): self
     {
         $this->getResource()->setRecipientTaxIdentificationNumbers($taxIdentificationNumbers);
 
         return $this;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function addRecipientTaxIdentificationNumber(TaxIdentificationNumber $taxIdentificationNumber)
+    public function addRecipientTaxIdentificationNumber(TaxIdentificationNumber $taxIdentificationNumber): self
     {
         $this->getResource()->addRecipientTaxIdentificationNumber($taxIdentificationNumber);
 
         return $this;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getRecipientTaxIdentificationNumbers()
+    public function getRecipientTaxIdentificationNumbers(): array
     {
         return $this->getResource()->getRecipientTaxIdentificationNumbers();
     }
 
-    /**
-     * @param AddressInterface $senderAddress
-     * @return $this
-     */
-    public function setSenderAddress(AddressInterface $senderAddress)
+    public function setSenderAddress(AddressInterface $senderAddress): self
     {
         $this->getResource()->setSenderAddress($senderAddress);
 
         return $this;
     }
 
-    /**
-     * @return AddressInterface
-     */
-    public function getSenderAddress()
+    public function getSenderAddress(): ?AddressInterface
     {
         return $this->getResource()->getSenderAddress();
     }
 
     /**
-     * @param string|null $senderTaxNumber
-     * @return $this
-     * @deprecated
+     * @deprecated Use setSenderTaxIdentificationNumbers() or addSenderTaxIdentificationNumber() instead.
      */
-    public function setSenderTaxNumber($senderTaxNumber)
+    public function setSenderTaxNumber(?string $senderTaxNumber): self
     {
         $this->getResource()->setSenderTaxNumber($senderTaxNumber);
 
         return $this;
     }
 
     /**
-     * @return string|null
-     * @deprecated
+     * @deprecated Use getSenderTaxIdentificationNumbers() instead.
```

Comment:
It honestly sounds like a good moment to get rid of deprecated things if we're updating to a new major version, don't you think?

Perhaps a quick `Find in Project` search could make it easy to get rid of these deprecated methods?

Reply:
While I would love to get rid of deprecated stuff, the scope of the current project is to support PHP 8. Since there are quite some things to deprecate, I think it's better to do this in another project and release another major version.